src/test/rustdoc-ui/intra-doc/disambiguator-mismatch.stderr

```diff
@@ -2,94 +2,138 @@ error: incompatible link kind for `S`
+   |
+LL | /// Link to [enum@S]
+   |              ^^^^^
 
 error: incompatible link kind for `S`
   --> $DIR/disambiguator-mismatch.rs:29:14
    |
 LL | /// Link to [trait@S]
-   |              ^^^^^^^ help: to link to the enum, prefix with `enum@`: `enum@S`
+   |              ^^^^^^^
    |
    = note: this link resolved to an enum, which is not a trait
+help: to link to the enum, prefix with `enum@`
+   |
+LL | /// Link to [enum@S]
+   |              ^^^^^
 
 error: incompatible link kind for `T`
   --> $DIR/disambiguator-mismatch.rs:34:14
    |
 LL | /// Link to [struct@T]
-   |              ^^^^^^^^ help: to link to the trait, prefix with `trait@`: `trait@T`
+   |              ^^^^^^^^
    |
    = note: this link resolved to a trait, which is not a struct
+help: to link to the trait, prefix with `trait@`
+   |
+LL | /// Link to [trait@T]
+   |              ^^^^^^
 
 error: incompatible link kind for `m`
   --> $DIR/disambiguator-mismatch.rs:39:14
    |
 LL | /// Link to [derive@m]
-   |              ^^^^^^^^ help: to link to the macro, add an exclamation mark: `m!`
+   |              ^^^^^^^^
    |
    = note: this link resolved to a macro, which is not a derive macro
+help: to link to the macro, add an exclamation mark
+   |
+LL | /// Link to [derive@m!]
+   |                      ^
 
 error: incompatible link kind for `s`
   --> $DIR/disambiguator-mismatch.rs:44:14
    |
 LL | /// Link to [const@s]
-   |              ^^^^^^^ help: to link to the static, prefix with `static@`: `static@s`
+   |              ^^^^^^^
    |
    = note: this link resolved to a static, which is not a constant
+help: to link to the static, prefix with `static@`
+   |
+LL | /// Link to [static@s]
+   |              ^^^^^^^
 
 error: incompatible link kind for `c`
   --> $DIR/disambiguator-mismatch.rs:49:14
    |
 LL | /// Link to [static@c]
-   |              ^^^^^^^^ help: to link to the constant, prefix with `const@`: `const@c`
+   |              ^^^^^^^^
    |
    = note: this link resolved to a constant, which is not a static
+help: to link to the constant, prefix with `const@`
+   |
+LL | /// Link to [const@c]
+   |              ^^^^^^
 
 error: incompatible link kind for `c`
   --> $DIR/disambiguator-mismatch.rs:54:14
    |
 LL | /// Link to [fn@c]
-   |              ^^^^ help: to link to the constant, prefix with `const@`: `const@c`
+   |              ^^^^
    |
    = note: this link resolved to a constant, which is not a function
+help: to link to the constant, prefix with `const@`
+   |
+LL | /// Link to [const@c]
+   |              ^^^^^^
 
 error: incompatible link kind for `c`
   --> $DIR/disambiguator-mismatch.rs:59:14
    |
 LL | /// Link to [c()]
-   |              ^^^ help: to link to the constant, prefix with `const@`: `const@c`
+   |              ^^^
    |
    = note: this link resolved to a constant, which is not a function
+help: to link to the constant, prefix with `const@`
+   |
+LL | /// Link to [const@c()]
+   |              ^^^^^^
```

Comment:
This is subtly incorrect: we're failing to remove the `()`. You have two options here: either use `multipart_suggestion`, which takes a `vec![(Span, String)]` of replacements, or for this case in particular, use the "whole span replacement" that we were already doing.

Reply:
Also note that `multipart_suggestion` when given a vec with a single element behaves the same as `span_suggestion` (and also has a `_verbose` counterpart), so you don't need to special case the two modes if you go down that route.

Reply:
It doesn't seem to have a `_verbose` counterpart?

Reply:
Ah! I added one, but in a PR that hasn't been merged yet. Disregard for now then.